[Week10] 전남대 17조 BE 10주차 리뷰 PR

Woojae-Jeong opened the pull request:
안녕하세요 멘토님~ 시작이 엊그제 같은데 벌써 10주차라니 믿겨지지가 않습니다 ㅜㅜ 
멘토님 덕분에 팀원들끼리 깊은 생각 공유를 할 수 있어서 좋았으며 많이 성장할 수 있었습니다! 감사드립니다

## 구현 사항
이번 주차는 주로 기능들 다듬고 고도화 하는 방향으로 진행하였습니다
- 일부 기능 테스트 코드 작성(주말동안 남은 기능들 작성 예정)
- 팀 비밀번호 null값 허용하도록 리팩토링
- 관리자 페이지 접근을 허용여부를 판단하기 위해 MemberDTO 에 role추가
- 회원가입 시 랜덤 닉네임 부여 기능 추가
- 인덱스 추가
- 팀 기능 고도화 및 책임 분산을 위한 일부 메소드 Team 엔티티 내부에 구현

## 질문 사항
- ExerciseHisoty는 BaseEntity를 상속 받아서 createdAt이라는 필드가 존재합니다 근데 ExerciseHistory가 생성되는 시점이 다음날 새벽 3시에 생성이 되므로 DB를 볼 때 10일날 기록 createdAt 값이 11일로 저장되는 문제점이 존재합니다. 그래서 저희가 의논을 해본결과 ExerciseHistory는 수정되는 일이 없으므로 그냥 BaseEntity를 상속하는 것보다 엔티티 내에 별도로 createdAt이라는 필드를 둬서 10일의 기록은 createAt값이 10일로 저장되게 하자 라고 의논이 되었습니다. 이렇게 하면 모든 엔티티가 BaseEntity를 상속하기로 했던 것과는 다르게 되서 일관성이 깨지게 되는데 다소 일관성이 깨지더라도 보기 편한게 우선이니 이렇게 정하였습니다. 
멘토님께서는 정답이 없고 팀원들끼리 의논해서 결정한 사항이면 그대로 가시면 좋다고 하셨는데 그래도 이 사항에 대해서 멘토님의 의견이 듣고싶습니다! 앞으로도 다른 프로젝트를 할 때 코드를 편리하게 짜기 위해서 일관성을 깨야하는 경우를 적잖이 마주칠텐데, 그 때 현명한 판단을 하고자 멘토님은 보통 어떻게 하시는지 여쭙고 싶습니다!

nimunsang commented on src/main/java/homeTry/chatting/service/ChattingService.java:
```diff
@@ -23,62 +22,55 @@
     private final TeamService teamService;
-    private final TeamMemberService teamMemberService;
+    private final MemberService memberService;
+    private final TeamMemberMappingService teamMemberMappingService;
```
Chatting은 TeamMember와 연관관계를 맺고 있으니, TeamMember에 대한 정보만을 알고 있습니다.
Chatting에서 Team 또는 Member에 대한 정보를 얻고자 한다면,, TeamMemberService를 단일 진입점으로 활용하면 어떨까요? 😲

nimunsang commented on src/main/java/homeTry/chatting/interceptor/StompInterceptor.java:
```diff
@@ -36,16 +42,21 @@ public Message<?> preSend(Message<?> message, MessageChannel channel) {
 
-            if (!jwtAuth.validateToken(token))
-                throw new InvalidChattingTokenException();
+            MemberDTO memberDTO;
```
StompInterceptor는 토큰 정보를 보고, 멤버가 존재하지 않으면,, Exception을 뱉도록 하는 것 같습니다.
그렇다면, stompInterceptor는 토큰이 bearerToken인지, 다른 토큰의 종류인지? 에 대해서는 아무런 관심사가 없습니다. BearerToken이 유효한지? 도 관심사가 없습니다. 단순히 현재 interceptor에 넘어온 토큰 정보를 보고, 우리 사용자라면 세션에 그 값을 넣는 것에 관심이 있습니다.
이렇게 각 클래스의 관심사들을 생각하며 적절하게 분리하고, 캡슐화 해보시면 재미있을 것 같아요 😄

nimunsang commented on src/main/java/homeTry/common/config/WebMvcConfig.java:
```diff
@@ -45,7 +45,7 @@ public void addArgumentResolvers(List<HandlerMethodArgumentResolver> argumentRes
     public void addCorsMappings(CorsRegistry registry) {
         registry.addMapping("/**")
-            .allowedOrigins("http://localhost:3000", "http://localhost:8080", "http://localhost:63342", "https://localhost:3000")
+            .allowedOriginPatterns("http://localhost:*", "https://localhost:*")
```
이제 최종적으로 배포도 수행하셔야 하니, 이렇게 localhost 로 host를 고정해두는 것 대신, 설정파일에 profile 별로 이 값을 설정하는 것은 어떨까요? (배포환경, 로컬 환경에 따라 달라지는 값은 yaml 파일에 작성해두시면 손쉽게 관리할 수 있습니다)
application-prod.yml 
application-dev.yml
처럼 나누어서 host를 설정할 수도 있을 것 같고, 
`spring.config.activate.on-profile` 로 나누셔도 좋을 것 같아요~

nimunsang commented on src/main/java/homeTry/exerciseList/event/ExerciseCreationEventListener.java:
```diff
@@ -0,0 +1,28 @@
+
+        ExerciseTime currentExerciseTime = new ExerciseTime(exercise);
+        exerciseTimeService.saveExerciseTime(currentExerciseTime);
+    }
```
오호~ 지난 주 피드백을 그대로 반영해주셨군요 😆
이러한 이벤트를 핸들링 할 때, 트랜잭션 범위를 어디까지 가져갈지? 에 대한 고민도 한 번쯤 해보시는 것이 좋을 것 같아요.
이벤트를 던지고 핸들링 하는것까지를 하나의 트랜잭션으로 가져갈 것인지? 아니면 이벤트를 던지는 것과, 이벤트를 핸들링하는 것을 별개의 트랜잭션으로 처리할 것인지? 에 대한 고민이에요~ (서비스의 요구사항에 따라 달라질 것이라고 생각해요)
그리고 그 방법에 대해서도 찾아보시면 분명히 도움이 되실거에요 😆

nimunsang commented on src/main/java/homeTry/common/entity/BaseEntity.java:
```diff
@@ -20,6 +20,8 @@ public abstract class BaseEntity {
     @LastModifiedDate
     private LocalDateTime updatedAt;
 
+    private boolean isDeprecated;
```
BaseEntity에 deprecated 필드를 통해 softDelete를 구현하고자 하신 것 같아요~ 
과연 모든 엔티티에 softDelete를 달아놓는 것이 부담이 되지는 않을지? 에 대해서도 많이 고민해보시고 이렇게 적용하셨을 것이라고 생각해요.
다만 현재 코드를 보면 softDelete를 적용한 엔티티가 있고, 아닌 엔티티도 존재하는데 모든 엔티티에 deprecated 필드를 적용하는 것은 사실 불필요한 필드가 추가되는 것이라고 생각해서,, softDelete를 적용하는 엔티티에만 각자 달아주는 것이 옳지 않나? 라는 생각이 있어요.

참고로, 직접 이렇게 softDelete를 관리하는 방법도 존재하지만, 하이버네이트 에서 제공해주는 `@SoftDelete` 라는 어노테이션도 존재해요. 직접 관리하는 방식과, 어노테이션을 활용한 방식에는 각각 장단점이 존재하기 때문에 한 번 그것에 대해서도 알아보셔도 좋을 것 같아요~ 요것도 정답은 없어요 😁

nimunsang commented on src/main/java/homeTry/exerciseList/model/entity/ExerciseTime.java:
```diff
@@ -8,12 +8,16 @@
+    name = "exercise_time",
+    indexes = {
+        @Index(name = "idx_exercise_time_exercise_start_time", columnList = "exercise_id, startTime"),
+        @Index(name = "idx_exercise_time_exercise_id", columnList = "exercise_id")
```
`exercise_id, start_time` 복합 인덱스를 활용하신다면,  exercise_id 단일 인덱스는 필요 없을 것 같아요!
exercise_id 로 이미 정렬되어 있을테니까요!

nimunsang commented on src/main/java/homeTry/member/service/MemberService.java:
```diff
@@ -9,39 +10,47 @@
-    public Long register(MemberDTO memberDTO) {
+    public MemberDTO register(KakaoMemberInfoDTO kakaoMemberInfoDTO) {
+
+        MemberDTO memberDTO = new MemberDTO(1L, kakaoMemberInfoDTO.email(),
```
MemberDTO에 임의로 id값을 1로 넘어주어야 하는 것은 부자연스러운 것 같아요 🥲 추후 리팩토링해보시면 좋을 것 같아요~

nimunsang commented on src/main/java/homeTry/member/service/MemberWithdrawService.java:
```diff
@@ -0,0 +1,31 @@
+        exerciseService.deleteAllExercisesByMemberId(memberId);
+        diaryService.deleteByMember(memberId);
+        memberService.withdrawMember(memberId);
+    }
```
이 동작은 반드시 트랜잭션으로 묶여야 할 것 같아요!
cascade 속성을 이용하시는 것도 좋을 것 같아요~

nimunsang commented on src/main/java/homeTry/member/utils/RandomNicknameGenerator.java:
```diff
@@ -0,0 +1,48 @@
+    };
+
+    public static String generateNickname() {
+        Random random = new Random();
```
난수를 생성할 때, `ThreadLocalRandom` 에 대해서도 알아보셔도 좋을 것 같아요~

nimunsang commented on src/main/java/homeTry/product/service/AdminProductService.java:
```diff
@@ -1,29 +1,49 @@
+    @Transactional
+    public void addProduct(ProductRequest request, MemberDTO memberDTO) {
+        // 관리자 권한 확인
+        verifyAdmin(memberDTO);
```
어드민만 사용할 수 있는 서비스이고, /admin 경로로 부터 사용됩니다.
이러한 인증 로직은 서비스 로직이 아닌, 더 앞단에서 공통으로 처리하는 것이 어떨까요?
또는, Controller 레벨에서 그냥 손쉽게 어노테이션 하나만으로 처리할 수도 있을 것 같고, 인터셉터에서 그냥 잡아버려도 괜찮겠네요.
추후 리팩토링 대상으로 포함해보셔도 좋을 것 같아요!

nimunsang commented on src/main/java/homeTry/product/service/ProductService.java:
```diff
@@ -51,9 +51,9 @@ private Slice<Product> getProductsByTagIds(List<Long> tagIds, Pageable pageable)
+    // 특정 상품 선택 시 해당 상품의 조회 수 증가
     @Transactional
-    public String incrementViewCountAndGetUrl(Long productId, MemberDTO memberDTO) {
+    public void incrementViewCount(Long productId, MemberDTO memberDTO) {
```
여러 요청이 동시에 이 메서드에 접근하면 어떻게 될까요?
이 조회수 증가 로직에 대한 **동시성 보장 방법**에 대해 반드시 고민해보세요~

nimunsang commented on src/main/java/homeTry/team/model/entity/Team.java:
```diff
@@ -82,13 +85,32 @@ public Optional<Password> getPassword() {
-        this.maxParticipants = new Participant(maxParticipants);
-        this.currentParticipants = new Participant(currentParticipants);
-        this.password = new Password(password);
+    public void decreaseParticipantsByWithdraw() {
```
이렇게 특정 숫자가 더해지고,, 빼지는 로직들에 대해서는 동시성 관리가 필요합니다. atomic한 연산을 구현하는 방법에 대해 알아보세요!

nimunsang reviewed (approved):
안녕하세요~ 😄 벌써 마지막 리뷰네요. 저도 믿기지 않네요..ㅎㅎ
그동안 정말 고생 많으셨습니다. 저도 매주 17조가 계속 발전하는 모습을 보며 즐거움을 느꼈어요. 감사합니다  😆
이번에도 리뷰 몇 가지 달아드렸으니, 한 번 확인해보시고, 팀원분들과 논의해보셔도 좋을 것 같아요~ 카테캠이 종료되겠지만,, 프로젝트를 함께 계속 발전시키는 것도 좋은 경험일 것입니다. 여러 프로젝트를 하는 것보다.. 하나의 프로젝트를 꾸준히 발전시키는 것이 중요합니다. 각자의 사정으로 함께 할 수 없다면, 개인적으로라도 실행해보세요!


질문을 살펴보자면,,
BaseEntity로 저장되는 createdAt, modifiedAt, createdBy, modifiedBy는 "데이터베이스" 에 저장/수정이 이루어 지는 것을 기록합니다. (Auditing 기능은 JPA 기능입니다)
하지만 ExerciseHistory가 새벽 3시에 생성되고,,,~ 는 우리 서비스의 비즈니스 로직입니다. 이러한 로직을 별개로 저장하는 것은 자연스러운 것이라고 생각합니다. "데이터베이스"에 저장되는 시점과는 별개로, 비즈니스 로직에서 생성된 시점을 분리하여 저장하는 것이 옳을 것 같습니다. (기존의 DB 저장 시점인 createdAt을 그대로 두고, 운동 기록 시간인 exercisedAt(?) 을 별도로)
이렇게 함으로써 시스템의 확장성이나 유지보수성을 높일 수 있습니다. 실제로 서비스를 운영하다보면 이러한 시간적인 정보를 활용하여 운영해야 할 경우가 종종 있습니다.
결론은,,  "비즈니스에서 활용하는 시간 정보" 와, "데이터베이스에 저장하는 시간 정보" 를 별개로 생각하시는 것이 좋습니다. 다만, 데이터베이스에 저장하는 시간 정보를 비즈니스에서 활용할 수는 있겠죠. 예를 들면,, 우리 서비스는 Order Entity가 데이터베이스가 저장되는 시점을 "주문 일시" 로 정할거야! 처럼요.
이 문제는 사실 어느정도 정답은 정해져 있는 것 같아서 분리하는 것을 권장드립니다 😆

> 앞으로도 다른 프로젝트를 할 때 코드를 편리하게 짜기 위해서 일관성을 깨야하는 경우를 적잖이 마주칠텐데, 그 때 현명한 판단을 하고자 멘토님은 보통 어떻게 하시는지 여쭙고 싶습니다!
- 현재 서비스와, 상황에 대한 최선의 방법이나 코드 컨벤션은 어느 정도 정해져 있다고 생각합니다. 이러한 방법들에 대한 일관성을 유지하며 코드를 짜는 것이 여러 사람들이 다같이 코드를 편리하게 작성할 수 있는 방법이라고 생각합니다. (물론 이러한 방법을 처음부터 잘 적용할 수 있냐? 그것은 또 매우 어려운 문제입니다..) 
지금 단계에서는 최선의 방법과 컨벤션을 배워가는 단계라, 일관성을 깨야 하는 경우가 적잖이 마주칠 것이라고 생각됩니다. 여러 시행착오를 통해 배워 나가는 부분이라, 열심히 학습하시고 부딪히면서 배우시는 것이,, 좋을 것 같습니다 🤣 
- 저도 아직 열심히 학습하는 단계라, 혼자 현명한 판단을 내리기는 꽤 어렵습니다. 훌륭한 동료분들과 함께 고민을 나누며 문제를 해결하는 것이 제 해결 방법인데요. 이번 프로젝트를 진행하며 팀원분들과 여러 소통을 나누어 보라고 계속 말씀드린 이유가 이것이기도 합니다. 혼자서는 못보던 문제가,, 머리를 모아서 다른 시각에서 바라보면 또 다른 것이 됩니다. 앞으로도 함께 진행할 프로젝트의 팀원분들과 꾸준히 소통하며 현명한 판단을 내리는 것이 최선의 방법이라고 생각합니다.



이제 정말 카테캠 마무리까지 얼마 남지 않았는데요, 정말 고생 많으셨고 앞으로도 화이팅입니다 🔥

hxunsc committed:
refactor: 컬럼명 명시 제거

rdme0 committed:
Merge pull request #231 from cliant/week10

[Week10] adminInterceptor 구현에 따른 Tag내 관리자 권한 확인 로직 제거

rdme0 committed:
Merge pull request #232 from hxunsc/feature/exercise

[Week10] 운동 저장 부분 수정 및 인터셉터 구현에 따른 상품 관리자 권한 확인 로직 제거

rdme0 committed:
feat: 채팅 응답에 채팅 날짜 필드 추가

rdme0 committed:
feat: 팀 탈퇴 기능 구현 (미완성)

rdme0 committed:
merge: 병합 충돌 해결

Woojae-Jeong committed:
refactor: 채팅 기록 삭제는 TeamWithdrawService의 deleteTeam 메소드에서 수행됌

Woojae-Jeong committed:
refactor: TeamWithdrawService 에서 멤버탈퇴라는 사실을 모르게 리팩토링

기존에 있던 메소드들은 멤버 탈퇴가 있어야지 기능을 수행한다는 점에 있어서 좋지 해당 서비스에 책임이 무거워지고 재사용성이 낮았다. 하지만 멤버탈퇴서비스에서 deleteTeam과 withdrawTeam 메소드를 적절히 호출해 줌으로써 재사용성도 높이고 책임도 멤버 서비스가 가지는 적절한 책임 분배가 이루어짐

Woojae-Jeong committed:
Merge pull request #233 from Woojae-Jeong/groupDevelop

[Week10] TeamWithdrawService 책임 분산으로 인한 리팩토링

rdme0 committed:
Merge branch 'Week10' of https://github.com/kakao-tech-campus-2nd-step3/Team17_BE into feature/chatting-week10

hxunsc committed:
refactor: valid 어노테이션 추가

rdme0 committed:
feat: FE 배포 서버 CORS Mapping 추가

rdme0 committed:
feat(Member): Member 엔티티의 비활성화 필드 추가

rdme0 committed:
feat: 회원 탈퇴 기능 완성

rdme0 committed:
feat: 회원 탈퇴 기능 완성

rdme0 committed:
feat(GlobalExceptionHandler): 예상치 못한 에러 시 서버에 로깅하는 기능 추가

rdme0 committed:
feat(GlobalExceptionHandler): 예상치 못한 에러 시 서버에 로깅하는 기능 추가

rdme0 committed:
Merge pull request #234 from rdme0/feature/chatting-week10

[Week10] 회원 탈퇴 기능 완성 & FE 배포 url 반영 & 에러 로깅 기능 추가

rdme0 committed:
refactor(INACTIVATED_MEMBER_EXCEPTION): 응답코드 400 -> 403

rdme0 committed:
Merge pull request #235 from rdme0/feature/chatting-week10

[Week10] 응답코드 리팩터링

rdme0 committed:
fix: CORS 에러 수정

rdme0 committed:
Merge pull request #236 from rdme0/feature/chatting-week10

fix: CORS 에러 수정

Commit:
refactor: Tag 관리자 페이지에 돌아가기 버튼 추가

hxunsc committed:
refactor: 운동 종료 로직 수정

해당 예외 발생 시 트랜잭션 롤백이 이루어지는 것을 방지하여 운동 상태가 isActive=false 의 상태가 저장되도록 함

hxunsc committed:
test: 운동 종료와 관련한 테스트 코드

hxunsc committed:
test: 상품 관리 페이지 관련 테스트코드

Woojae-Jeong committed:
Merge pull request #237 from cliant/week11

[Week10] 태그 관리 페이지내에 돌아가기 버튼 추가

hxunsc committed:
remove: 사용하지 않는 컨트롤러 제거

Woojae-Jeong committed:
Merge pull request #238 from kakao-tech-campus-2nd-step3/Week10

[Week10] Week10 -> Develope  최종

Commit:
refactor: 디렉토리 대소문자 변경

Commit:
chore: swagger의존성 추가

Woojae-Jeong committed:
refactor: 내가 가입한 그룹 조회 시 탈퇴한 그룹은 뜨지 않게끔 jqpl 수정

Commit:
refactor: 대소문자 변경

Woojae-Jeong committed:
chore: Team 도메인 관련 Exception들 badRequestException 패캐지로 이동 및 GlobalExceptionHandler의 패키지 이동

Woojae-Jeong committed:
Merge pull request #239 from cliant/week11

[Week11] 패키지 구조 통일 및 swagger 의존성 추가

Woojae-Jeong committed:
Merge pull request #240 from hxunsc/feature/exercise

[Week11] 운동 시간 관련 버그 해결

Woojae-Jeong committed:
Merge pull request #241 from Woojae-Jeong/Week11Develop

[Week11] team 예외들 패키지 이동, 내가 가입한 그룹 조회에 관한 jpql 수정, GlobalExceptionHandler 의 패키지 이동

Commit:
refactor: 변수명 명시

rdme0 committed:
feat: 관리자 페이지 구현 및 path variable 버그 해결

rdme0 committed:
Merge pull request #242 from cliant/week11

[Week11] refactor: 변수명 명시

rdme0 committed:
merge: 병합 충돌 해결

rdme0 committed:
remove: 사용하지 않는 html 태그 제거

rdme0 committed:
Merge pull request #243 from rdme0/feature/chatting-week11

[Week11] 관리자 페이지 구현

hxunsc committed:
refactor: 상품 조회 메서드 수정

hxunsc committed:
feat: 스웨거 문서화 추가

hxunsc committed:
refactor: 불필요한 예외 및 인자 삭제

rdme0 committed:
style: 코드 정리

rdme0 committed:
style: 주석 제거

hxunsc committed:
refactor: 관리자 메인페이지로 가는 버튼 추가

rdme0 committed:
feat: 배포 / 로컬 개발 프로필 설정

rdme0 committed:
feat: 배포할 때 prod 프로필을 사용하도록 구현

rdme0 committed:
style: 코드 정리

hxunsc committed:
remove: 사용하지 않는 에러 삭제

hxunsc committed:
chore: data.sql 수정

Woojae-Jeong committed:
refactor: 랭킹 조회 시 운동기록이 없는 유저는 랭킹에서 제외

만약 유저도 운동을 하지 않은 채로 랭킹을 조회하는 경우 0으로 랭킹을 0의 값으로 줌

Woojae-Jeong committed:
refactor: 사용하지 않은 기능 삭제

Woojae-Jeong committed:
refactor: TeamMemberMapping 정보를 찾을 때 isDeprecated값이 false인 정보만 조회

Woojae-Jeong committed:
refactor: 가입하지 않은 팀 조회 시 탈퇴했던 팀도 조회되도록 jpql 수정

Woojae-Jeong committed:
refactor: TeamMemberMapping 정보 조회 시 softDelete 된 정보는 제외시키고 조회하도록 jpql 수정

hxunsc committed:
chore: data.sql 실제 가격 반영

Woojae-Jeong committed:
refactor: 랭킹 조회 시 팀에 소속되지 않은 멤버의 조회요청인 경우 거절하도록 리팩토링

Woojae-Jeong committed:
refactor: 랭킹 조회 시 팀에 소속되지 않은 멤버의 조회요청인 경우 거절하도록 리팩토링

Woojae-Jeong committed:
Merge pull request #245 from Woojae-Jeong/develop

[Week11] 랭킹 조회, 팀 조회 버그 수정

rdme0 committed:
refactor: 관리자 기능 더 스마트 해지도록 리팩터링

1. 로컬 스토리지에 토큰이 있는지 여부에 따라 로그인/관리자 권한 페이지 버튼 렌더링
2. MemberDTO를 이용해서 role에 따라 관리자 권한 페이지를 다르게 렌더링
3. 예외 처리 강화
4. 좀 더 직관적인 ui 디자인

rdme0 committed:
Merge pull request #246 from rdme0/feature/chatting-week11

[Week11] 배포/로컬 프로필 설정 & 관리자 페이지 강화

hxunsc committed:
feat: 상품 수정 기능 추가

hxunsc committed:
feat: 상품 수정 기능 html 추가

rdme0 committed:
refactor: kakao redirect url 변경

rdme0 committed:
Merge pull request #248 from rdme0/feature/chatting-week11

[Week10] kakao redirect url 변경

rdme0 committed:
Merge pull request #247 from hxunsc/feature/exercise

[Week11] 상품 관련 수정 및 스웨거 적용

rdme0 committed:
refactor: 오타 수정

rdme0 committed:
Merge pull request #249 from rdme0/feature/chatting-week11

[Week11] 오타 수정

hxunsc committed:
refactor: 상품 수정 html 수정

rdme0 committed:
style: 반응형 -> px

rdme0 committed:
Merge pull request #250 from rdme0/feature/chatting-week11

[Week11] 반응형 -> px로 리팩터링

hxunsc committed:
refactor: 상품 이름 최대 글자수 변경

hxunsc committed:
refactor: uri 변경

hxunsc committed:
refactor: html uri 변경

rdme0 committed:
refactor: onclick 리팩터링

hxunsc committed:
refactor: onclick 속성에 getRequestWithToken 사용

Commit:
refactor: 태그 관리 uri 변경

Commit:
chore: 사용하지 않는 태그관리 controller 삭제

Commit:
refactor: adminPage.html의 onclick uri 수정

rdme0 committed:
Merge pull request #251 from hxunsc/feature/exercise

[Week11] 상품 edit 버튼 안눌리는 버그 해결 및 관리자 상품 uri 변경

rdme0 committed:
Merge pull request #252 from cliant/week11

[Week11] 태그관리 uri 수정 및 사용하지 않는 컨트롤러 삭제

rdme0 committed:
merge: conflict 해결

rdme0 committed:
refactor: 상품 삭제 기능 RESTful 하도록 리팩터링

rdme0 committed:
refactor(MemberService): 더 깔끔한 코드로 리팩터링

hxunsc committed:
refactor: 초과 시간 관련 로직 수정

8시간 제한, 12시간 제한에 대해 예외 반환을 하지 않고 내부적으로 시간만 저장하지 않도록 수정

hxunsc committed:
test: 수정된 코드에 의해 테스트 코드 수정

hxunsc committed:
remove: 사용하지 않는 에러 삭제

Commit:
test: 테스트 코드 수정

hxunsc committed:
refactor: 반환 시 시작시간 추가

hxunsc committed:
test: 테스트 코드 수정

rdme0 committed:
fix: 관리자 코드 틀릴 때 올바르지 않은 응답 버그 해결

rdme0 committed:
refactor: 상품 기능 더 RESTful 하도록 리팩터링

rdme0 committed:
Merge pull request #253 from rdme0/feature/chatting-week11

[Week11] 상품 삭제 더 RESTful 하도록 리펙터링

rdme0 committed:
Merge pull request #255 from hxunsc/feature/exercise

[Week11] 시간 관련 버그 수정

rdme0 committed:
Merge pull request #256 from cliant/week11

[Week11] Swagger 적용 및 redirectURI 수정

hxunsc committed:
refactor: 하루 총 12시간 넘을 경우에 대해 수정

하루 운동시간이 12시간 초과시에는 11시간 59분 59초까지만 저장되도록 수정

hxunsc committed:
test: 변경된 AdminProductController에 의한 테스트코드 수정

rdme0 committed:
refactor: CORS Mapping 수정

Woojae-Jeong committed:
feat: 팀 기능 테스트 구현

Woojae-Jeong committed:
Merge pull request #257 from hxunsc/feature/exercise

[Week11] 12시간 초과 운동에 대한 처리 수정 및 테스트 코드 수정

Woojae-Jeong committed:
Merge pull request #258 from Woojae-Jeong/Week11

feat: 팀 기능 테스트 구현

Woojae-Jeong committed:
refactor: 팀 이름으로 팀 조회에 대한 jpql 수정

요청으로 들어온 단어가 들어있는 팀을 정상적으로 조회되게 수정

Woojae-Jeong committed:
refactor: 시간이 같은 팀원은 같은 랭킹이 되도록 랭킹 수정 및 0초인 팀원들도 랭킹에 보이게 수정

Woojae-Jeong committed:
feat: Team 기능 Swagger 작성

Woojae-Jeong committed:
Merge pull request #259 from Woojae-Jeong/Week11

[Week11] 팀 조회 및 팀 랭킹 관련 버그 수정, Team 기능 Swagger 작성

rdme0 committed:
Merge branch 'Week11' of https://github.com/kakao-tech-campus-2nd-step3/Team17_BE into feature/chatting-week11

hxunsc committed:
chore: 운동관련 스웨거 수정

hxunsc committed:
chore: 상품 관리자 스웨거 추가

Commit:
chore: Parameter hidden 설정

hxunsc committed:
chore: 스웨거 수정

hxunsc committed:
style: 어노테이션 위치 수정

hxunsc committed:
chore: 스웨거 설명 수정

Commit:
chore: 어노테이션 순서 변경

Commit:
chore: Parameter 히든설정 변경

Commit:
chore: 불필요한 import 삭제

rdme0 committed:
refactor: uri : /api/oauth/login/ssr -> /oauth/login/ssr

api가 아니기 때문

rdme0 committed:
build: swagger 버전 최신화

rdme0 committed:
feat: /docs -> /swagger-ui/index.html 로 리다이렉트 하는 기능 추가

rdme0 committed:
docs: Member, Chatting, Auth, Admin 기능 Swagger docs 작성

rdme0 committed:
Merge pull request #260 from hxunsc/feature/exercise

[Week11] 스웨거 수정

rdme0 committed:
merge: 병합 충돌 해결

rdme0 committed:
refactor: 디렉토리 이동

rdme0 committed:
Merge pull request #262 from rdme0/feature/chatting-week11

[Week11] Swagger 작성, url 수정, /docs -> /swagger-ui/index.html 리다이렉트 구현

rdme0 committed:
Merge pull request #261 from cliant/week11

[Week11] 어노테이션 순서 변경

rdme0 committed:
test: 팀 탈퇴 테스트 구현

rdme0 committed:
Merge pull request #263 from rdme0/feature/chatting-week11

[Week11] 팀 탈퇴 테스트 구현

rdme0 committed:
refactor: 클래스 간 결합력 최소화

rdme0 committed:
Merge pull request #264 from rdme0/feature/chatting-week11

[Week11] 클래스 간 결합력 최소화

hxunsc committed:
feat: 상품 수정 js 구현

hxunsc committed:
refactor: 상품 수정 html 수정

hxunsc committed:
refactor: url 백틱을 따옴표로 수정

rdme0 committed:
refactor(Stomp): 관심사의 분리, 예외 응답 리팩터링

rdme0 committed:
Merge pull request #265 from rdme0/feature/chatting-week11

[Week11] 관심사의 분리 & 예외 응답 리팩터링

rdme0 committed:
refactor: 랜덤 닉네임 생성기 -> thread-safe 하게 리팩터링

rdme0 committed:
Merge pull request #267 from rdme0/feature/chatting-week11

[Week11] 랜덤 닉네임 생성기 -> thread-safe 하게 리팩터링

hxunsc committed:
refactor: url 변경

hxunsc committed:
refactor: 상품 수정 html, js 수정

Commit:
refactor: 상품 태그 관리 기능 리팩토링

Commit:
refactor: 팀 태그 관리 기능 리팩토링

Commit:
chore: 불필요한 XML 네임스페이스 선언 삭제

Commit:
refactor: 논리삭제된 태그 이름에 대한 중복검사 제외

Commit:
chore: tag에 swagger 적용

hxunsc committed:
chore: 스웨거 그룹화 및 헤더 정보 포함

rdme0 committed:
Merge pull request #266 from hxunsc/feature/exercise

[Week11] 상품 관리자 페이지 내 버그 수정

rdme0 committed:
Merge pull request #268 from cliant/week11

[Week11] 태그 관리 기능 리팩토링 및 swagger 적용

Woojae-Jeong committed:
refactor: Team 컨트롤러 어노테이션 순서 변경

Woojae-Jeong committed:
docs: 잘못된 데이터 값 수정

Woojae-Jeong committed:
Merge pull request #269 from Woojae-Jeong/Week11

[Week11] Swagger 어노테이션 순서 변경 및 data.sql 오타 수정

rdme0 committed:
refactor: 상품 버튼 이미지로 리팩터링

rdme0 committed:
Merge pull request #270 from rdme0/feature/chatting-week11

[Week11] 상품 버튼을 이미지로 리팩터링

rdme0 committed:
refactor: 상품 edit 디자인 리팩터링

rdme0 committed:
Merge pull request #271 from rdme0/feature/chatting-week11

[Week11] 상품 edit 디자인 리팩터링

rdme0 committed:
refactor: 예외 응답 리팩터링

hxunsc committed:
chore: 상품 데이터 추가

rdme0 committed:
feat: 회원 탈퇴할때 kakao unlink 하는 기능 구현

rdme0 committed:
Merge pull request #272 from hxunsc/feature/exercise

[Week11] 상품 테스트 데이터 추가

rdme0 committed:
Merge pull request #273 from rdme0/feature/chatting-week11

[Week11] 회원 탈퇴할 때 카카오 로그인 unlink 하는 기능 구현

rdme0 committed:
build: MySQL 커넥터 추가

rdme0 committed:
refactor: MySQL 마이그레이션

rdme0 committed:
refactor: MySQL 마이그레이션

rdme0 committed:
Merge pull request #274 from rdme0/feature/chatting-week11

[Week11] MySQL로 마이그레이션

rdme0 committed:
refactor: 관리자 페이지 디자인 개선

rdme0 committed:
Merge pull request #275 from rdme0/feature/chatting-week11

[Week11] 관리자 페이지 디자인 개선

hxunsc committed:
refactor: 상품 수정 시 save 문 제거

hxunsc committed:
refactor: 불필요한 save 문 삭제

Woojae-Jeong committed:
Merge pull request #276 from hxunsc/feature/exercise

[Week11] 관리자 상품 수정 버그 해결

Woojae-Jeong committed:
refactor: 팀 가입 및 팀 탈퇴 시 동시성 제어가 되도록 수정

TeamWithdrawService -> TeamJoinAndWithdrawService 로 바꾸면서 팀 가입과 팀 탈퇴 시 synchronized 키워드를 통한 동기화가 적용된 메소드를 호출함으로써 값에 대한 수정 발생 시 동시성 제어가 이루어지게 함

Woojae-Jeong committed:
Merge pull request #277 from Woojae-Jeong/Week11

[Week11] 팀 가입 및 팀 탈퇴 시 동시성 제어가 되도록 수정

rdme0 committed:
feat: 개발 환경 - h2 / 배포 환경 - MySQL 기능 추가

rdme0 committed:
Merge pull request #278 from rdme0/feature/chatting-week11

[Week11] 개발환경 - H2 DB, 배포환경 - MySQL 사용하도록 구현

Commit:
test: 태그 테스트 수정

rdme0 committed:
feat: 개발 환경 - data-dev.sql 기능 추가

rdme0 committed:
Merge pull request #280 from rdme0/feature/chatting-week11

[Week11] 개발 환경에 data-dev.sql 사용하는 기능 추가

rdme0 committed:
Merge pull request #279 from cliant/week11

[Week11] 태그 테스트 수정

rdme0 committed:
refactor: 기본 프로필 dev로 변경

rdme0 committed:
Merge pull request #281 from rdme0/feature/chatting-week11

[Week11] 기본 프로필 dev로 변경

rdme0 committed:
fix: data.sql을 schema로 설정했던 버그 수정

rdme0 committed:
Merge pull request #282 from rdme0/feature/chatting-week11

[Week11] data.sql을 schema로 설정했던 버그 수정

hxunsc committed:
test: test코드 active profile을 dev로 설정

hxunsc committed:
test: 테스트 코드 트랜잭션 제거

Woojae-Jeong committed:
Merge pull request #283 from hxunsc/feature/exercise

[Week11] 테스트 코드 수정

rdme0 committed:
fix: 채팅이 시간순으로 조회가 안되는 버그 해결

rdme0 committed:
Merge pull request #284 from rdme0/feature/chatting-week11

[Week11] 채팅 시간 순으로 조회 안되는 버그 해결

Woojae-Jeong committed:
feat: markAsActivated() 메소드 구현

팀 탈퇴 후 다시 팀에 가입 시 기존에 존재했던 TeamMemberMapping 데이터의 isDeprecated 값을 true로 바꿔주기 위해 구현

Woojae-Jeong committed:
refactor: TeamMemberMapping을 추가할 때 이미 Deprecated 된 동일한 데이터가 있으면 다시 활성화

Woojae-Jeong committed:
feat: TeamMemberMapping 을 조회할 때 isDeprecated 값이 true인 것과 false인 데이터를 모두 조회해오는 jpa 메소드 구현

Woojae-Jeong committed:
refactor: TeamMemberMapping 데이터를 추가하기 위해서 존재하는 지 조회해올 때 Activated 된 값도 조회해오도록 수정

Woojae-Jeong committed:
Merge pull request #285 from Woojae-Jeong/Week11

[Week11] TeamMemberMapping 을 추가할 때 Deprecated된 값들도 같이 조회해오도록 수정

hxunsc committed:
refactor: 상품 수정 로직 수정

세터를 이용하도록 수정

rdme0 committed:
feat: 채팅 메세지 길이 500자 제한하는 기능 구현

rdme0 committed:
Merge pull request #287 from rdme0/feature/chatting-week11

[Week11] 채팅 메세지 길이 500자 제한하는 기능 구현

Woojae-Jeong committed:
feat: 이미 존재하는 TeamMemberMapping 데이터에 대해서 새로운 등록 요청 시 isDeprecated 값을 false로 정상적으로 전환되게 수정

Woojae-Jeong committed:
feat: 팀 설명 길이 제한 최대 255글자 구현

Woojae-Jeong committed:
Merge pull request #288 from Woojae-Jeong/Week11

[Week11] TeamMemberMapping 서비스 버그 수정 및, 팀 설명 길이 제한 구현

hxunsc committed:
refactor: 운동 전체시간 제한 로직 수정

hxunsc committed:
chore: 주석 수정

hxunsc committed:
refactor: 불필요한 메서드 분리 수정

hxunsc committed:
chore: 에러코드 번호 수정

rdme0 committed:
Merge pull request #289 from hxunsc/feature/exercise

[Week11] 상품 관리자 수정 및 전체 운동 시간 제한 수정

Woojae-Jeong committed:
refactor: 필요없는 태그 삭제

Woojae-Jeong committed:
refactor : 보기 좋게 에러 위치 변경

Woojae-Jeong committed:
Merge pull request #290 from Woojae-Jeong/develop

[Week11] 필요 없는 태그 데이터 삭제 및 에러 위치 변경

Commit:
refactor: 일기 메모 최대 길이 수정

Commit:
refactor: 예외 메세지 수정

cliant committed:
chore: upload erd

Commit:
chore: 이미지경로 수정

rdme0 committed:
Merge pull request #291 from cliant/week11

[Week11] 일기 유효성 검사 수정

hxunsc committed:
refactor: 상품 관리 페이지에서 다음 페이지로 못가던 버그 해결

rdme0 committed:
docs: readme 파일 테스트

cliant committed:
docs: Update README.md

hxunsc committed:
docs: 폴더 구조 토글키를 이용하여 보도록 수정

hxunsc committed:
docs: API 명세서 링크 수정

rdme0 committed:
refactor: 채팅 조회 디폴트 값 내림차순으로 리팩터링

rdme0 committed:
Merge pull request #292 from hxunsc/feature/exercise

[Week11] 상품 관리 페이지에서 다음 페이지로 못가던 버그 해결

rdme0 committed:
Merge pull request #293 from rdme0/feature/chatting-week11

[Week11] 채팅 조회 디폴트 값 내림차순으로 리팩터링

rdme0 committed:
docs: Update README.md

rdme0 committed:
docs: update README.md

hxunsc committed:
Update README.md

hxunsc committed:
docs: Update README.md

cliant committed:
docs: 일기, 태그에 관한 기능 개발 주안점 작성

hxunsc committed:
docs: 개발기간 및 회의록 링크 추가

rdme0 committed:
docs: Update README.md

rdme0 committed:
docs: update README.md

hxunsc committed:
docs: Update README.md

rdme0 committed:
docs: 회원 탈퇴 부분 작성

rdme0 committed:
docs: Update README.md

hxunsc committed:
chore: 안쓰는 import 문 및 메서드 정리

rdme0 committed:
docs: 들여쓰기 되어있는거 수정

Woojae-Jeong committed:
Merge pull request #294 from hxunsc/feature/exercise

[Week11] 안쓰는 import 문 및 메서드 정리

Woojae-Jeong committed:
Update README.md

그룹 설명 추가

hxunsc committed:
docs: Update README.md

cliant committed:
docs: 개발주안점에 이모지 추가

hxunsc committed:
docs: Update README.md

rdme0 committed:
docs: README 채팅 부분 작성

rdme0 committed:
docs: README 채팅 부분 작성

hxunsc committed:
docs: 개발 주안점 이모지 뒤 공백 추가

rdme0 committed:
Merge pull request #295 from kakao-tech-campus-2nd-step3/Week11

[Week11] Develope <- Week11

rdme0 committed:
Merge pull request #296 from kakao-tech-campus-2nd-step3/Develope

[Week11] Master <- Develope

cliant committed:
Update README.md

cliant committed:
Update README.md

cliant committed:
Update README.md

cliant committed:
Update README.md

cliant committed:
Update README.md

cliant committed:
Update README.md

rdme0 committed:
Merge pull request #297 from kakao-tech-campus-2nd-step3/Week11

[Week11] Develope <- Week11

rdme0 committed:
Merge pull request #298 from kakao-tech-campus-2nd-step3/Develope

[Week11] Master <- Develope
